lib/oidcstrategy.js

```diff
@@ -69,18 +68,22 @@ const NONCE_MAX_AMOUNT = 10;
 const NONCE_LIFE_TIME = 3600; // second
 
 function makeProfileObject(src, raw) {
+  var emails = [];
+  if (src.upn)
+    emails = [src.upn];
```

Comment:
For AAD, upn is always in the email format like 'someone@sometenant.onmicrosoft.com'